esp-rtos/esp_config.yml

```diff
@@ -8,3 +8,13 @@ options:
     constraints:
       - type:
           validator: positive_integer
+
+  - name: sw-task-overflow-detection
+    description: 'Enable software-based stack overflow detection. The stack guard value and offset is based on esp-hal configuration.'
+    default:
+      - value: false
+
+  - name: hw-task-overflow-detection
+    description: 'Enable hardware-based stack overflow detection. The stack watermark is based on the esp-hal stack-guard-offset configuration.'
```

Comment:
I've elected to not use the esp-hal config here - the more we have, the more tightly the crates are coupled. Since the `stack-guard-*` options are currently unstable, we really shouldn't rely on them in esp-rtos at all (we risk breaking compatibility in the future).

While the stack protector is in the process of [getting stabilised](https://github.com/rust-lang/rust/pull/146369), it isn't stable yet.